static/js/metrics-cardinality.js

```diff
@@ -0,0 +1,178 @@
+        onGridReady: function (params) {
+            params.api.sizeColumnsToFit();
+        }
+    };
+
+    var tagPairsGridOptions = {
+        columnDefs: [
+            { headerName: 'Key', field: 'key', filter: true, resizable: true },
+            { headerName: 'Value', field: 'value', filter: true, resizable: true },
+            { headerName: 'Number of Series', field: 'numSeries', filter: true, resizable: true }
+        ],
+        rowData: [],
+        rowHeight: 44,
+        headerHeight: 32,
+        defaultColDef: {
+            cellClass: 'align-center-grid',
+
+            resizable: true,
+        },
+        enableCellTextSelection: true,
+        suppressScrollOnNewData: true,
+        suppressAnimationFrame: true,
+        onGridReady: function (params) {
+            params.api.sizeColumnsToFit();
+        }
+    };
+
+    var tagKeysValuesGridOptions = {
+        columnDefs: [
+            { headerName: 'Key', field: 'key', filter: true, resizable: true },
+            { headerName: 'Number of Unique Values', field: 'numValues', filter: true, resizable: true }
+        ],
+        rowData: [],
+        rowHeight: 44,
+        headerHeight: 32,
+        defaultColDef: {
+            cellClass: 'align-center-grid',
+
+            resizable: true,
+        },
+        enableCellTextSelection: true,
+        suppressScrollOnNewData: true,
+        suppressAnimationFrame: true,
+        onGridReady: function (params) {
+            params.api.sizeColumnsToFit();
+        }
+    };
+
+    new agGrid.Grid(document.getElementById('tagKeysGrid'), tagKeysGridOptions);
+    new agGrid.Grid(document.getElementById('tagPairsGrid'), tagPairsGridOptions);
+    new agGrid.Grid(document.getElementById('tagKeysValuesGrid'), tagKeysValuesGridOptions);
+
+    function fetchData(startEpoch, endEpoch) {
+        // Fetch total unique series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/series-cardinality', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch })
+        })
+            .then(response => response.json())
+            .then(data => {
+                totalUniqueSeries.textContent = data.seriesCardinality;
+            });
+
+        // Fetch tag keys with most series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/tag-keys-with-most-series', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch, limit: 10 })
+        })
+            .then(response => response.json())
+            .then(data => {
+                tagKeysGridOptions.api.setRowData(data.tagKeys);
+            });
+
+        // Fetch tag pairs with most series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/tag-pairs-with-most-series', {
```

Comment:
I see this error for this call.
<img width="404" alt="image" src="https://github.com/siglens/siglens/assets/71771131/83ec4139-3862-4424-a671-8b7bdb3d5485">
Please take a look at the request body to see if it is correct or if there is any other issue.

Reply:
@sunitakawane Can you please review this PR? I am getting an error that Divyanshu is not seeing. 
I tried deleting the data directory, but I'm still getting the error.